obinnaeye/quiz-generator
# Ui unit test setup Nextjs

# Pull Request: Set Up Basic Jest Testing Framework

## Overview

This pull request implements the initial setup for the Jest testing framework in our Next.js application. A basic test has been created to verify that the testing environment is functioning correctly.

## Changes Made

- Added a new test file located in `__tests__/simple.test.ts`.
- Implemented a simple test case that checks if the expression `true` evaluates to `true`.

### File Structure

client/
     src/
         app/
             __tests__/ 
                       │ └── simple.test.ts 

##Setup
1. To set up Jest, install jest and the following packages as dev dependencies:
`pnpm install -D jest jest-environment-jsdom @testing-library/react @testing-library/jest-dom`

2. Generate a basic Jest configuration file by running the following command:
`pnpm create jest@latest`
This will take you through a series of prompts to setup Jest for your project, including automatically creating a `jest.config.ts|js` file.
3. your `jest.config.ts` should look like this:
![image](https://github.com/user-attachments/assets/9eb80f7c-ac26-4e8d-8655-7e50e9eebdc6)
 
and jest.setup.ts
![image](https://github.com/user-attachments/assets/e0afed1c-5554-43a5-a12c-ca267b5df084)

4. add `"test:watch": "jest --watch" ` to your package.json file
5. create __tests__ folder where all the test file will be 

As a test i created simple.test.ts file run `pnpm test`
and this is the output
![image](https://github.com/user-attachments/assets/4541afca-186b-420b-b8a0-deb4e2b193bd)


## Purpose

The purpose of this PR is to:

1. **Verify Jest Installation**: Ensure that the Jest testing framework is correctly set up and can run tests in our project.
2. **Establish a Testing Foundation**: Provide a simple test case as a foundation for future test cases that will be added as the application develops.

## Testing Instructions

To verify that the test setup is working correctly, please follow these steps:

1. Pull the latest changes from this branch.
2. Navigate to the client directory of this project in your terminal.
6. Run the following command to execute the tests:

   ```bash
   pnpm test
